### PR TITLE
Add k4LCIOReader.

### PR DESCRIPTION
**Pull request by mirguest**

### CODE REVIEW

File: packages/k4lcioreader/package.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    variant('cxxstd',
+            default='17',
+            values=('14', '17'),
+            multi=False,
+            description='Use the specified C++ standard when building.')
+
+
+    depends_on('lcio')
+    depends_on('podio')
```
Comment (vvolkl) [inline packages/k4lcioreader/package.py:25]:
```suggestion
    depends_on('podio@0.12:')
```

There's been some breaking changes, most likely it won't work with the older ones

File: packages/key4hep-stack/package.py [hunk 2]
```diff
@@ -225,6 +225,7 @@ class Key4hepStack(BundlePackage, Key4hepPackage):
     #######################################################
     depends_on("cepcsw")
     k4_add_latest_commit_as_dependency("cepcsw", "cepc/cepcsw", when="@master")
+    depends_on("k4lcioreader")
```
Comment (vvolkl) [inline packages/key4hep-stack/package.py:229]:
```suggestion
    k4_add_latest_commit_as_dependency("cepcsw", "cepc/cepcsw", when="@master")
    
    depends_on("k4lcioreader")
    k4_add_latest_commit_as_dependency("k4lcioreader", "key4hep/k4LCIOReader", when="@master")

```

**Comment by mirguest** [comment]
Hi @vvolkl,
Thank you for your review.
Tao

**Comment by vvolkl** [comment]
Thanks for adding this package!

**Pull request merged**